Model/CustomerInformation.php

```diff
@@ -0,0 +1,123 @@
+<?php
+
+namespace NoFraud\Checkout\Model;
+
+use NoFraud\Checkout\Api\CustomerInformationInterface;
+use Magento\Framework\Webapi\Rest\Request;
+use Magento\Customer\Api\CustomerRepositoryInterface;
+use Magento\Customer\Model\Logger;
+use Magento\Sales\Model\ResourceModel\Order\CollectionFactory;
+use Magento\Customer\Model\Customer;
+use Magento\Vault\Api\PaymentTokenManagementInterface;
+
+class CustomerInformation implements CustomerInformationInterface
+{
+    protected $request;
+    protected $customerRepository;
+    protected $customerLog;
+    protected $orderCollectionFactory;
+    protected $customerCollection;
+    protected $paymenttokenmanagement;
+
+    public function __construct(
+        Request                         $request,
+        CustomerRepositoryInterface     $customerRepository,
+        Logger                          $customerLog,
+        CollectionFactory               $orderCollectionFactory,
+        Customer                        $customerCollection,
+        PaymentTokenManagementInterface $paymenttokenmanagement
+    ) {
+        $this->request                = $request;
+        $this->customerRepository     = $customerRepository;
+        $this->customerLog            = $customerLog;
+        $this->orderCollectionFactory = $orderCollectionFactory;
+        $this->customerCollection     = $customerCollection;
+        $this->paymenttokenmanagement = $paymenttokenmanagement;
+    }
+
+    public function getCustomerInformation()
+    {
+        try {
+            $body       = $this->request->getBodyParams();
+            $customerId = $body['data']['customer_id'];
+
+            // Retrieve customer information by ID
+            $customer      = $this->customerRepository->getById($customerId);
+            $id            = $customer->getId();
+            $email         = $customer->getEmail();
+            $createdAt     = $customer->getCreatedAt();
+            $createdDate   = date('m/d/Y', strtotime($createdAt));
+            $lastLoginAt   = $this->customerLog->get($customerId);
+            $lastLoginDate = date('m/d/Y', strtotime($lastLoginAt->getLastLoginAt()));

```

Comment:
SonarCloud suggests using a class constant for m/d/Y

Reply:
Hi @soleilcot , I've added the recommenedation. see here https://github.com/Nofraud/nofraud_magento_checkout/pull/41/files